### PR TITLE
Fixes #5459 and Atmos Tech Traitor Item 

### LINKED ISSUE #5459: Alt+Click Doesn't Work Correctly For Silicons
As a vox/diona/human/skeleton, if you alt+click an object, it pulls up the list of everything that's on that tile.
As a silicon, it only pulls up the list if you alt+click on the floor. Alt+clicking an object does nothing. This means that if the tile has a 32x32 object, it's impossible to click on the floor, and so impossible to get the list for that tile. In general, it just makes things more difficult on cluttered spaces because you have to pixel hunt.

### PR DESCRIPTION
**Pull request by Kurfursten**
- Contortionist's Jumpsuit - 7 TC, Job Specific: Atmos Tech
- Allows you to navigate the vents and scrubbers as if you were a monkey or MoMMI
- Does not protect you from atmosphere or temperature -- plan accordingly with your configuration
- Allows you to carry gloves, glasses, mask, ID, shoes, and pocket items while ventcrawling; but not anything else (e.g.: belt, backpack, helmet, hardsuit, in-hand)

Needs general spritework, currently using the gimpy looking suit.

Now you can finally break into the vault and armory by going through the vents, James Bond style.
- Fully tested

fixes #5459

### CODE REVIEW

**Comment by Arthurdentist** [comment]
If wearing clown shoes whilst using this suit means you still honk every step in the pipe, then I will love this item forever.

**Comment by Exxion** [comment]
>Ventcrawl clown shoes honking

#5166
Please oh please work

**Comment by FudgePucker** [comment]
My problem with atmos techs current antag item is that it is pointless to the job.
Its a shotgun, a murder boner tool, disguised as an item that you would have no reason to have on you anyways.
Engineer gets the perfect job specific antag item, and also one of my favorites, the power glove.
All those rounds of trying to make autistic super engines pay off, not only that, but an engineer wearing a pair of yellow gloves isn't out of the ordinary.

My ideas for the atmos tech traitor item was backwards pumps (Or a custom made RPD that lets you make false devices, that isn't a gas filter its a gas mixer), meaning you could sabotage atmospherics where when they try to fix it, it only gets worse. Or a special conversion kit for canisters, that let you set up the valve to open from a device, or/and weaken the canister's structure so it can burst (Or rupture you EVIL EVIL BASTARD (Just fucking double the bomb cap for canister bombs.))

But this jumpsuit idea works as well, you have to set up atmospherics perfectly so that you don't take pressure damage. Only problem I see is it just being used to snatch your objective/captain's spare ID and nothing else.

**Comment by NigglyWiggly** [comment]
>icon_state = "darkholme"

Hey buddy I think you got the wrong vent, the leather club's two blocks down.

**Comment by Probe1** [comment]
@FudgePacker is says right in the OP 

>Allows you to carry gloves, glasses, mask, ID, shoes, and pocket items while ventcrawling; but not anything else (e.g.: belt, backpack, helmet, hardsuit, in-hand)

You can't carry the shotgun in your hands or on your back.  Or a backpack to carry it in.  It's a very, very specialized item to purchase.  I had doubts at first glance but after considering it, it's pretty strategic and it isn't a hella OP item.

**Comment by FudgePucker** [comment]
Have it also be available in one of the bundles, but have it be Mario's outfit, his hat, and a fire flower that's just a water flower filled with plasma.

@Probe1 
Eh, you have to remember that engineers traitor item is literally disintegrate when standing on top of a power wire and having the engine wired to grid, and can be worn with a hardsuit on.
>Atmos tech not in hardsuit? VALIDS DETECTED.

Also
>Plus if you know someones using it just flush waste with superheated gas and laugh and laugh and laugh.

Its an atmos tech only traitor item, they have atmospheric access, they just need to set distro to 101 kpa so that they can survive. Waste is going to be at 0 or 100000000. So all you really need to do is set distro to higher pressure to kill the vent crawlers. Superheating distro just to get rid of a vent crawler will lead to, BWANK, Did you really just sabotage distro to get valids?

**Comment by Probe1** [comment]
I said waste, not distro.  

Metagaming that an atmostech not in a hardsuit must be a traitor is ridiculous.  

Atmos techs have a modified flare gun traitor item already that won't be removed. It sets victims on fire and does the same amount of damage as an egun.  It can also be loaded with shotgun shells.

>Eh, you have to remember that engineers traitor item is literally disintegrate

Power gloves do not ash people.  They have a lengthy cooldown and max out at 70 damage.  They are also unavailable to atmos techs so they are kind of irrelevant to a discussion about an atmos tech traitor item.

**Comment by FudgePucker** [comment]
@Probe1 
>Power gloves do not ash people or in any way gib them.

Fun fact: (Last time I used them, so unless there was a recent change.) With enough power in grid (Why are you playing engineer if you can't put 10 deloreans into the power grid?), when you use the power gloves it leaves nothing but ash and a remains (like if you sacrificed someone with a rune.)
Like zap and they are removed. (And I don't see where they nerfed it, but I haven't gotten an engineer antag roll in so long.)
Luckily for me it also sometimes leaves the ID, I picked up the captains ID after I showed him the real power of autism.
>I said waste, not distro

Why the hell would you crawl though waste, the place where super hot plasma and other bullshit goes to get vented to space and at anytime could kill you, instead of distro, where pressure is controlled by gas pump and usually will not change from what you set it to, putting on more gas pumps to 'help' the station to keep it at a safe pressure for yourself if there is a breach.
>Atmos techs have a modified flare gun traitor item

Yea, and its pointless.
Don't believe me?
Name one use of the flare gun outside of self defense, a very shitty reason to carry the normal flare gun (I'm sure security won't beat in your face for having it for that reason.(Also its shit.)), there isn't one, why the hell would you need to shoot a flare for? Okay now name one use for insulated gloves. Oh that's right, its the thing that all engineers wear, and all the assistants, and all of the cargo members, and the power gamin HoP, and everyone who can get their hands on a pair. Atmos tech with a flare gun sticks out like a skinhead at a gay pride parade, engineer with insulated gloves would be more suspicious if he wasn't wearing a pair.
>They are also unavailable to atmos techs so they are kind of irrelevant to a discussion about an atmos tech traitor item.

Atmos techs and engineers have the exact same access. They are pretty much the exact same job besides different job titles and different traitor items.

**Comment by Probe1** [comment]
Fudge I don't know what's up your ass about this but you're pulling a lot of things from it.  I made my last post with a test server up to confirm what I wrote was true.  Believe whatever you want but please stick to the pull request.  And please actually read the damn pull requests changelog.

**Comment by PJB3005** [comment]
\>Roundstart atmos tech traitor.
\>Buy contortionist's jumpsuit.
\>No captain, steal spare ID 30 sec into round.

There's a balance issue here.

**Comment by Probe1** [comment]
Is that really different from buying an emag

**Comment by PJB3005** [comment]
with an emag people light be able to actually catch you, because you're still incredibly obvious

**Comment by Probe1** [comment]
dunno.  you have a clear advantage with the atmos jumpsuit but it's limited to atmos techs.  you can see above you and know whether its safe to exit the pipe and leave the same way.  

with an emag you can just wait for no one to be around, waltz in, hope the cappin isn't there, and climb into disposals for a clean getaway.  I think our regular non antag assistants and clowns prove that stealing the spare is a trivial exercise without the help of any traitor tools.  

i'm not overly worried about what happens when no one secures the spare.  it is bound to go missing whether via this item or a multitool or a welder and wrench.

**Comment by Kurfursten** [comment]
NFD?

**Comment by Arthurdentist** [comment]
#PJB3005 Emag has greater utility because it can do far more then be an easy way in and out of a place. This item is great for stealthy entry but you can't use it to do all the things an emag can do on top of opening doors.

File: code/_onclick/ventcrawl.dm [hunk 1]
```diff
@@ -103,7 +112,7 @@ var/list/ventcrawl_machinery = list(/obj/machinery/atmospherics/unary/vent_pump,
 					if(!client)
 						return
 
-					if(contents.len && !isrobot(src))
+					if((contents.len && !isrobot(src))&&!allowed_carry)
```
Comment (clusterfack) [inline code/_onclick/ventcrawl.dm]: Remove the isrobot check and give mommis the allowed carry argument

File: code/_onclick/ventcrawl.dm [hunk 2]
```diff
@@ -103,7 +112,7 @@ var/list/ventcrawl_machinery = list(/obj/machinery/atmospherics/unary/vent_pump,
 					if(!client)
 						return
 
-					if(contents.len && !isrobot(src))
+					if((contents.len && !allowed_carry)
```
Comment (Kurfursten) [inline code/_onclick/ventcrawl.dm]: Missing )

**Comment by Kurfursten** [comment]
Conflict resolved; I guess I thought this PR was merged and never really checked to see what was the deal

**Comment by clusterfack** [comment]
Conflicts gone, but compile failed

**Comment by Kurfursten** [comment]
Truly the lesson is to always compile before pushing

**Comment by ComicIronic** [comment]
You completely overwrote my PR!

**Comment by Kurfursten** [comment]
Had to rebase due to conflicts, push it again

**Comment by Intigracy** [comment]
Slap a "fixes #5459" somewhere in the OP.

**Comment by Kurfursten** [comment]
Still waiting on this NFD to merge

**Comment by d3athrow** [comment]
In the future do not put fixes in a feature PR. keep it atomic

**Comment by ComicIronic** [comment]
That was me, actually, and it was to do with how I rewrote the ventcrawl system to work better.

**Pull request merged**